src/RevCore.cc

```diff
@@ -281,19 +256,19 @@ bool RevCore::InitTableMapping() {
     NameToEntry.insert( std::pair<std::string, unsigned>( ExtractMnemonic( InstTable[i] ), i ) );
     if( !InstTable[i].compressed ) {
       // map normal instruction
-      EncToEntry.insert( std::pair<uint32_t, unsigned>( CompressEncoding( InstTable[i] ), i ) );
+      EncToEntry.insert( std::pair<uint64_t, unsigned>( CompressEncoding( InstTable[i] ), i ) );
       output->verbose(
         CALL_INFO,
         6,
         0,
-        "Core %" PRIu32 " ; Table Entry %" PRIu32 " = %s\n",
+        "Core %" PRIu32 " ; Table Entry %" PRIu64 " = %s\n",
         id,
         CompressEncoding( InstTable[i] ),
         ExtractMnemonic( InstTable[i] ).data()
       );
     } else {
       // map compressed instruction
-      CEncToEntry.insert( std::pair<uint32_t, unsigned>( CompressCEncoding( InstTable[i] ), i ) );
+      CEncToEntry.insert( std::pair<uint64_t, unsigned>( CompressCEncoding( InstTable[i] ), i ) );
```

Comment:
CompressCEncoding returns a uint32_t, here we are upcasting to a uint64_t, which is probably fine, but should we just use uint64_t everywhere to be consistent?

Reply:
We use `uint64_t` whenever it is necessary, but we don't make any unnecessary changes, such as in the compressed instruction code. `uint32_t` is automatically upconverted to `uint64_t` when adding to or searching the tables. I don't think that storing tables in `uint64_t` and storing or searching for `uint32_t` keys is bad.